File: src/main/java/org/tech6/services/usuario/UsuarioService.java
package org.tech6.services.usuario;

import org.tech6.dto.usuario.UsuarioDTO;
import org.tech6.dto.auth.validar.ValidarResponseDTO;
import org.tech6.model.usuario.Usuario;
import org.tech6.repository.usuario.UsuarioRepository;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.transaction.Transactional;

import java.util.List;
import java.util.UUID;

@ApplicationScoped
public class UsuarioService {

    private final UsuarioRepository userRepository;

    public UsuarioService(UsuarioRepository userRepository) {
        this.userRepository = userRepository;
    }

    public List<Usuario> buscarUsuarios() {
        return userRepository.listAll();
    }

    public List<UsuarioDTO> inValidUsers() {
        List<Usuario> usuarios = userRepository.findByValidoFalse();
        return usuarios.stream().map(usuario -> new UsuarioDTO(
                usuario.id.toString(),
                usuario.nome + " " + usuario.sobreNome,
                usuario.cpf.getValorFormatado(),
                usuario.email,
                usuario.telefone.getValorFormatado(),
                usuario.role.name(),
                usuario.valido == true ? "Sim" : "Não",
                usuario.cadastro)).toList();
    }

    @Transactional
    public ValidarResponseDTO validar(UUID id) {
        int linhasAfetadas = userRepository.validarUsuario(id);

        if (linhasAfetadas == 0) {
            return new ValidarResponseDTO(true, "Não existe usuario com id: " + id);
        }

        return new ValidarResponseDTO(false, "Usuário atualizado com sucesso!");
    }
}

File: src/main/kotlin/com/fatec/destino/services/usuario/UsuarioService.kt
package com.fatec.destino.services.usuario

import com.fatec.destino.dto.auth.validar.ValidarResponseDTO
import com.fatec.destino.dto.usuario.UsuarioDTO
import com.fatec.destino.model.usuario.Usuario
import com.fatec.destino.repository.usuario.UsuarioRepository
import jakarta.transaction.Transactional
import org.springframework.data.repository.findByIdOrNull
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Service
import java.util.*

@Service
class UsuarioService(private val userRepository: UsuarioRepository) {
    fun buscarUsuarios(): List<Usuario?> {
        return userRepository.findAll().orEmpty()
    }

    fun inValidUsers(): ResponseEntity<List<UsuarioDTO>> {
        val usuarios = userRepository.findByValidoFalse() ?: emptyList()

        val dtos = usuarios.filterNotNull().map { u ->
            UsuarioDTO(
                u.id.toString(),
                "${u.nome} ${u.sobreNome}",
                u.cpf.getvalorFormatado(),
                u.email,
                u.telefone?.valorFormatado,
                u.perfil.name,
                if (u.valido) "Sim" else "Não",
                u.cadastro
            )
        }
        return ResponseEntity.ok(dtos)
    }

    @Transactional
    fun validar(id: UUID): ResponseEntity<ValidarResponseDTO> {
        val linhasAfetadas = userRepository.validarUsuario(id)

        if (linhasAfetadas == 0) {
            return ResponseEntity.ok()
                .body(ValidarResponseDTO(true, "Não existe usuario com id: $id"))
        }

        return ResponseEntity.ok()
            .body(ValidarResponseDTO(false, "Usuário atualizado com sucesso!"))
    }
}
